### DISCUSSION TITLE
Is DKVP not safe as an input format?

### DISCUSSION THREAD

**Discussion by trantor** [johnkerl/miller#1078] [Q&A] [answered]
Hello @johnkerl and everyone.
I've been trying to use the DKVP output of one `mlr` as input for a second `mlr` command. However I noticed that field values containing a `,` character do not lead to the field value being quoted, hence that comma is treated as a field separator for the second `mlr` run in input.
Since I do not have control on which characters may or may not appear in the field values, that makes DKVP useless a format as it stands in my use case. Which is annoying since it's one of the most readable for displaying purposes.
Is that a known thing or a bug? Checked both with 6.3.0 and 5.10.2.

**Comment by johnkerl** [comment, marked as answer]
DKVP is completely non-escaped and requires fields without `,` or `=` in them -- or more generally without whatever the `FS` and `PS` are

One option is use less-common `FS` and `PS`, like `\001` and `\002`

But what it really needs is double-quoting (like CSV has) or some escaping (like TSV has) -- which is https://github.com/johnkerl/miller/issues/266 -- and which (sadly) has been open a loooong time. :(

**Reply by trantor** [reply]
Thanks John

**Reply by trantor** [reply]
However shouldn't this be noted in the docs @johnkerl ?

**Reply by johnkerl** [reply]
@trantor sure, I can update
